datafusion/sql/src/unparser/expr.rs

```diff
@@ -1256,6 +1273,30 @@ mod tests {
             ),
             (col("need-quoted").eq(lit(1)), r#"("need-quoted" = 1)"#),
             (col("need quoted").eq(lit(1)), r#"("need quoted" = 1)"#),
+            (
+                interval_month_day_nano_lit(
+                    "1 YEAR 1 MONTH 1 DAY 3 HOUR 10 MINUTE 20 SECOND",
+                ),
+                r#"INTERVAL '0 YEARS 13 MONS 1 DAYS 3 HOURS 10 MINS 20.000000000 SECS'"#,
+            ),
```

Comment:
I removed some tests because I think we use the formatter in `arrow-cast` now. We can assume the formatting behaviors are tested there. We don't need to keep so many tests here.